scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/DaemonEnvironment.java

```diff
@@ -48,6 +54,30 @@ public StartTask startTaskFor(ContainerInstance i) {
         .build();
   }
 
+  public List<StopTask> stopTaskFor(final List<Task> tasks) {
+    final List<String> tasksToStop =
+        tasks
+            .stream()
+            .filter(this::unmatchesTask)
+            .map(Task::getTaskDefinitionArn)
+            .collect(Collectors.toList());
+    return tasksToStop
+        .stream()
+        .map(
+            t ->
+                StopTask.builder()
+                    .clusterName(environment.getClusterName())
+                    .task(t)
+                    .reason("replaceAfterTerminate")
+                    .build())
+        .collect(Collectors.toList());
+  }
+
+  public boolean unmatchesTask(Task t) {
```

Comment:
From the name, I would expect `unmatchesTask(t) == !matchesTask(t)` for all `t`. This doesn't seem to be the case, so perhaps it's time to give both of these methods more descriptive names?